src/zenml/integrations/github/code_repositories/github_code_repository.py

```diff
@@ -62,6 +63,20 @@ class GitHubCodeRepositoryConfig(BaseCodeRepositoryConfig):
 class GitHubCodeRepository(BaseCodeRepository):
     """GitHub code repository."""
 
+    @classmethod
+    def validate_config(cls, config: Dict[str, Any]) -> None:
+        """Validate the code repository config.
+
+        This method should check that the config/credentials are valid and
+        the configured repository exists.
+
+        Args:
+            config: The configuration.
+        """
+        code_repo = cls(id=uuid4(), config=config)
+        # Try to access the project to make sure it exists
+        _ = code_repo.github_repo
```

Comment:
what happens if it doesnt exist

Reply:
It raises an exception that tells users that the repo doesn't exist. Or that something else went wrong, depending on the result of the response I guess